pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -243,6 +244,8 @@ private enum LineageUpdateType {
   private final LineageManager _lineageManager;
   private final RebalancePreChecker _rebalancePreChecker;
   private TableSizeReader _tableSizeReader;
+  private final ExecutorService _executorService;
+  private HttpClientConnectionManager _connectionManager;
```

Comment:
as mentioned, let's not create this and pass it in, but directly create this in the code where we need it

Reply:
afaik, most classes reuse the executor/connectionmgr created by ControllerStarter.

Restlet classes use dependency injection to get access to those objects, thus looks cleaner. For PinotHelixResourceManager, looks like the convention is to create register...() methods to get access to those objects, so it looks fine to me to use register...() methods here.